src/Profile.js

```diff
@@ -1,6 +1,56 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import { propTypes } from "react-bootstrap/esm/Image";
+import { user } from "./data/user.js";
 
 const Profile = () => {
+  var edit_mode = false;
+  const [d_interests, setd_interests] = useState(user.d_interests);
+  const [m_interests, setm_interests] = useState(user.m_interests);
+  var will = true;
+  var anon = false;
+
+  const remove_dtag = (e) => {
+    e.preventDefault();
+    var name = e.target.innerText.substring(2);
+    setd_interests(d_interests.filter((item) => item != name));
+  };
+
+  const remove_mtag = (e) => {
+    e.preventDefault();
+    var name = e.target.innerText.substring(2);
+    setm_interests(m_interests.filter((item) => item != name));
+  };
+
+  const add_dtag = (e) => {
+    //e.preventDefault();
+    var new_tag = document.getElementById("add-dtag").value;
+    setd_interests([...d_interests, new_tag]);
+  };
+
+  const add_mtag = (e) => {
+    //e.preventDefault();
+    var new_tag = document.getElementById("add-mtag").value;
```

Comment:
in javascript, using the `let` (for values that  can change) or `const` (for values that shouldn't change) keywords are better than using `var`. it's less ambiguous I believe

Reply:
Gotcha! I'll change it then.

Reply:
Gotcha! I'll change it then.